### PR TITLE
chore(saved-aggregations-queries): Add tracking for the new feature COMPASS-5508

### PR DESCRIPTION
**Pull request by gribnoysup**
Adds new tracking events and modifies existing ones for the new feature

### CODE REVIEW

File: packages/compass-aggregations/src/modules/index.js [hunk 1]
```diff
@@ -685,7 +685,6 @@ export const getPipelineFromIndexedDB = (pipelineId) => {
     fs.readFile(file, 'utf8', (error, data) => {
       if (!error) {
         const pipe = JSON.parse(data);
-        track('Aggregation Opened', { num_stages: pipe.pipeline.length });
```
Comment (gribnoysup) [inline packages/compass-aggregations/src/modules/index.js:688]: We use the same action to open the aggregation from the "my queries" screen, so moved the tracking out of here so we don't get multiple events for the same thing

File: packages/compass-saved-aggregations-queries/src/hooks/use-grid-filters.tsx [hunk 2]
```diff
@@ -92,54 +100,70 @@ function useSearchFilter(): [React.ReactElement, string] {
...
+    onCollectionSelect,
+    selectedCollection,
+  ]);
+
+  const selectState = useMemo(() => {
+    return {
+      database: selectedDatabase,
+      collection: selectedCollection,
+    };
+  }, [selectedDatabase, selectedCollection]);
```
Comment (gribnoysup) [inline packages/compass-saved-aggregations-queries/src/hooks/use-grid-filters.tsx:166]: Refactored this a bit to make sure that object is updated only if one of the values changed

File: packages/compass-saved-aggregations-queries/src/stores/delete-item.ts [hunk 3]
```diff
@@ -3,6 +3,9 @@ import { FavoriteQueryStorage } from '@mongodb-js/compass-query-history';
 import { PipelineStorage } from '@mongodb-js/compass-aggregations';
 import type { ThunkAction } from 'redux-thunk';
 import type { RootState } from '.';
+import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
```
Comment (mabaasit) [inline packages/compass-saved-aggregations-queries/src/stores/delete-item.ts:6]: do we need to `npm install @mongodb-js/compass-logging` in this package?

File: packages/compass-saved-aggregations-queries/src/stores/delete-item.ts [hunk 4]
```diff
@@ -76,6 +79,17 @@ export const deleteItemConfirm = (): ThunkAction<
     if (!item) {
       return;
     }
+
+    track(
+      item.type == 'aggregation'
+        ? 'Aggregation Deleted'
```
Comment (mabaasit) [inline packages/compass-saved-aggregations-queries/src/stores/delete-item.ts:85]: Just remembered that in `pipelineStorage.js` we do tracking when we delete an aggregation. Should we move these to the respective classes?

Comment (gribnoysup) [reply packages/compass-saved-aggregations-queries/src/stores/delete-item.ts:85]: Ah, missed that somehow, thanks. I don't think we can do this from these stores for the same reasons I moved `Aggregation Opened` outside of the action into a different place. If you do it from store, then you need to pass `screen` information to the store also and this just leaks implementation details that the store shouldn't care about I think

Comment (gribnoysup) [reply packages/compass-saved-aggregations-queries/src/stores/delete-item.ts:85]: Done in 5111259cf7cfb3404c74958a9585af733972eb55 but please tell me if the reasoning makes sense to you, we can talk more about it and then change it if needed

File: packages/compass-saved-aggregations-queries/src/stores/delete-item.ts (hunk 3 above)
Comment (gribnoysup) [reply packages/compass-saved-aggregations-queries/src/stores/delete-item.ts:6]: Yes, fixed, thanks

**Review by mabaasit — APPROVED** [review]

**Pull request merged**